app/src/main/java/kr/ac/kaist/team888/mylittlefont/MainActivity.java

```diff
@@ -179,4 +187,83 @@ public void onClick(DialogInterface dialog, int which) {
 
     dialog.show();
   }
+
+  private void saveFeatureValues() {
+    final EditText nameInput = new EditText(this);
+    nameInput.setSingleLine();
+
+    AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
+    dialogBuilder.setTitle(R.string.feature_save_name);
+    dialogBuilder.setView(nameInput);
+    dialogBuilder.setPositiveButton(R.string.feature_save, null);
+    dialogBuilder.setNegativeButton(R.string.feature_save_cancel,
+        new DialogInterface.OnClickListener() {
+          @Override
+          public void onClick(DialogInterface dialog, int which) { }
+        });
+
+    AlertDialog dialog = dialogBuilder.create();
+
+    // When the save button is clicked.
+    dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+      @Override
+      public void onShow(final DialogInterface dialog) {
+        Button button = ((AlertDialog) dialog).getButton(AlertDialog.BUTTON_POSITIVE);
+        button.setOnClickListener(new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+            String name = nameInput.getText().toString();
+            // Name should not be empty.
+            if (name.isEmpty()) {
+              Toast.makeText(getApplicationContext(), R.string.feature_save_empty_name,
+                  Toast.LENGTH_SHORT).show();
+              return;
+            }
+            new DatabaseOpenHelper(getApplicationContext())
```

Comment:
Why don't we hold DBhelper in MainActivity? Loading multiple helper object every time we save seems inefficient.